src/layer/tile/GridLayer.js

```diff
@@ -444,7 +445,8 @@ L.GridLayer = L.Layer.extend({
 
 	_getTiledPixelBounds: function (center) {
 		var map = this._map,
-		    scale = map.getZoomScale(map.getZoom(), this._tileZoom),
+		    mapZoom = map._animatingZoom ? Math.max(map._animateToZoom, map.getZoom()) : map.getZoom(),
+		    scale = map.getZoomScale(mapZoom, this._tileZoom),
```

Comment:
@yohanboniface You're missing fractional zoom. In those cases, 0.5 < scale < 2. See the `debug/tests/tile-bounds.html` file.